tests/queries/0_stateless/02149_read_in_order_fixed_prefix.sql

```diff
@@ -0,0 +1,21 @@
+DROP TABLE IF EXISTS t_read_in_order;
+
+CREATE TABLE t_read_in_order(date Date, i UInt64, v UInt64)
+ENGINE = MergeTree ORDER BY (date, i);
+
+INSERT INTO t_read_in_order SELECT '2020-10-10', number % 10, number FROM numbers(100000);
+INSERT INTO t_read_in_order SELECT '2020-10-11', number % 10, number FROM numbers(100000);
+
+SELECT date, i FROM t_read_in_order WHERE date = '2020-10-11' ORDER BY i LIMIT 10;
```

Comment:
Maybe also worth adding max_rows_to_read limit?

Reply:
BTW using `max_rows_to_read` `.sh` test can be rewritten to `.sql` too.

Reply:
As I remember `max_rows_to_read` doesn't work with `optimize_read_in_order`, because number of rows to read is estimated by selected ranges and we check it at query analysis stage before execution. In general case, when there exists `WHERE` clause it's impossible to estimate number of rows more precisely and it's not done.

Reply:
>As I remember max_rows_to_read doesn't work with optimize_read_in_order, because number of rows to read is estimated by selected ranges and we check it at query analysis stage before execution.

You are right, completely forgot about this.

Reply:
`max_rows_to_read` should work from the statistics of actually read rows.

```
milovidov@milovidov-desktop:~/work/ClickHouse/tests/queries/0_stateless$ clickhouse-local 
ClickHouse local version 21.13.1.1.

milovidov-desktop :) SET max_rows_to_read = 1000000

SET max_rows_to_read = 1000000

Query id: 5d8b9e6d-2c60-4fa7-a30f-a14ac6867a70

Ok.

0 rows in set. Elapsed: 0.000 sec. 

milovidov-desktop :) SELECT count() FROM system.numbers

SELECT count()
FROM system.numbers

Query id: 9eb383c3-e0b0-4754-ab26-1c8506a0860c


0 rows in set. Elapsed: 0.112 sec. 

Received exception:
Code: 158. DB::Exception: Limit for rows or bytes to read exceeded, max rows: 1.00 million, current rows: 1.05 million: While executing Numbers. (TOO_MANY_ROWS)
```